src/libs/web3/useWeb3Network.tsx

```diff
@@ -24,7 +26,16 @@ export const useWeb3Network = () => {
   const [networkError, setNetworkError] = useState<string>();
 
   const switchNetwork = useCallback(async () => {
-    await connector.activate(1);
+    try {
+      await connector.activate(config.network.chainId);
+    } catch (e: any) {
+      // If the chainId is unrecognized by wallet
+      if (e.code === 4902) {
```

Comment:
Dug deeper on this to avoid problems with other wallets. The behaviour of web3-react is to attempt to do `connector.activate(desiredChainIdOrChainParameters)` with the provided `chainId` number if `typeof desiredChainIdOrChainParameters === number` or `desiredChainIdOrChainParameters.chainId`. If not possible, it already catches error code 4902 for both coinbase and metamask inject wallets and checks if networkParams is of type `AddEthereumChainParameter` and if so, adds it to the wallet. This means it saves the step to add the network if it successfully uses the provided chainId, and there's no need for us to catch error code 4902.

So we can simplify the behaviour here by just passing `await connector.activate(getChainInfo());`

Because injected Coinbase Wallet is working, it has been added as an option to connect to Sei Network. Unfortunately couldn't test it locally because of how `process.env` is defined in the CoinbaseSDK and didn't want to waste time doing so, but it works in the cloudflare deployed page. Let me know if you want to get it working locally too.

Reply:
let open an issue for it and discuss next week what the best steps forward for this are

Reply:
https://github.com/bancorprotocol/carbon-app/issues/1208